fabric_cf/actor/fim/fim_helper.py

```diff
@@ -340,66 +344,29 @@ def get_interface_sliver_mapping(ifs_node_id: str, slice_graph: ABCASMPropertyGr
         Finds Peer Interface Sliver and parent information upto Network Node
         @param ifs_node_id node id of the Interface Sliver
         @param graph Slice ASM
+        @param itype Interface Type
         @returns Interface Sliver Mapping
         """
-        peer_cp_node_id = None
-
-        candidates = graph.get_first_and_second_neighbor(node_id=ifs_node_id, rel1=ABCPropertyGraph.REL_CONNECTS,
-                                                         node1_label=ABCPropertyGraph.CLASS_Link,
-                                                         rel2=ABCPropertyGraph.REL_CONNECTS,
-                                                         node2_label=ABCPropertyGraph.CLASS_ConnectionPoint)
-
-        if len(candidates) == 0:
-            return None
-        if len(candidates) != 1:
-            raise Exception(f"Connection point is only expected to connect to one other"
-                            f"connection point, instead connects to {len(candidates)}")
-        peer_cp_node_id = candidates[0][1]
-
-        clazzes, node_props = graph.get_node_properties(node_id=peer_cp_node_id)
-
-        # Peer Connection point maps to Interface Sliver
-        # Build Interface Sliver
-        peer_ifs = FimHelper.build_ifs_from_props(node_props=node_props)
-
-        return peer_ifs
-
-    @staticmethod
-    def get_interface_sliver_by_component_id_local_name(component_id: str, bqm: ABCCBMPropertyGraph,
-                                                        local_name: str) -> InterfaceSliver or None:
-        """
-        Find Interface Sliver matching a local name when the parent component Id is provided
-        @param component_id node id of the parent component
-        @param bqm Broker Query Model
-        @param local_name local Name of the connection point
-        @returns Interface Sliver
-        """
-        cp_ids = bqm.get_first_and_second_neighbor(node_id=component_id,
-                                                   rel1=ABCPropertyGraph.REL_HAS,
-                                                   node1_label=ABCPropertyGraph.CLASS_NetworkService,
-                                                   rel2=ABCPropertyGraph.REL_CONNECTS,
-                                                   node2_label=ABCPropertyGraph.CLASS_ConnectionPoint)
-
-        cp = None
-        if len(cp_ids) == 0:
-            return None
-        elif len(cp_ids) == 1:
-            cp_id = cp_ids[0][1]
-            _, cp_props = bqm.get_node_properties(node_id=cp_id)
-            cp = FimHelper.build_ifs_from_props(node_props=cp_props)
-        else:
-            for c in cp_ids:
-                _, cp_props = bqm.get_node_properties(node_id=c[1])
-                if local_name in cp_props[ABCPropertyGraph.PROP_NAME]:
-                    cp = FimHelper.build_ifs_from_props(node_props=cp_props)
-                    break
-
-        return cp
+        result = []
+        candidates = graph.find_peer_connection_points(node_id=ifs_node_id)
```

Comment:
I made the change here to handle peers based on InterfaceType. 

Assumptions:
In ASM on finding a peer for a ServicePort, there will always be one peer.
In BQM, finding a peer for a Dedicated/Share Port of Itype TrunkPort, there will always be one peer.

Reply:
I believe these assumptions are correct. The only time you have more than one peer is for a TrunkPort in CBM or BQM when it is mapped to one or more FacilityPorts via the same Link object.